### PR TITLE
Fixed type bugs

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

File: sql/timetype.go [hunk 1]
```diff
@@ -120,17 +120,22 @@ func (t timespanType) ConvertToTimespanImpl(v interface{}) (timespanImpl, error)
 			minutes := absValue / 100
 			seconds := absValue % 100
 			if minutes <= 59 && seconds <= 59 {
+				microseconds := (seconds*microsecondsPerSecond) + (minutes * microsecondsPerMinute)
```
Comment (zachmu) [inline sql/timetype.go:123]: time.Duration does all these magnitude conversions for you, should use them unless you have very good reasons not to

https://stackoverflow.com/questions/41503758/conversion-of-time-duration-type-microseconds-value-to-milliseconds

**Review by zachmu — APPROVED** [review]
LG, just a few comments

File: sql/numbertype.go [hunk 2]
```diff
@@ -50,6 +50,15 @@ var (
 	Float32 = MustCreateNumberType(sqltypes.Float32)
 	// Float64 is a floating point number of 64 bits.
 	Float64 = MustCreateNumberType(sqltypes.Float64)
+
+	// decimal that represents the max value an uint64 can hold
+	dec_uint64_max = decimal.NewFromInt(9223372036854775807).Mul(decimal.NewFromInt(2).Add(decimal.NewFromInt(1)))
+	// decimal that represents the max value an int64 can hold
+	dec_int64_max = decimal.NewFromInt(9223372036854775807)
+	// decimal that represents the min value an int64 can hold
+	dec_int64_min = decimal.NewFromInt(-9223372036854775808)
```
[the remark this answers is not in the document]
Comment (Hydrocharged) [reply sql/numbertype.go:59]: For the `dec_uint64_max` var, `NewFromInt` only accepts `int64`, so the max `uint64` won't work.

File: sql/timetype.go (hunk 1 above)
Comment (Hydrocharged) [reply sql/timetype.go:123]: I remember running into issues using `time.Duration` before. I think it had to do with interpreting numbers as a string and whatnot, and converting that to a `time.Duration` took more code/effort than the current solution. It's something I could revisit in the future, but probably not worth completely rewriting the file for now.

**Pull request merged**